### PR TITLE
Run npm in pip install, remove gulp

### PR DESCRIPTION
**Pull request by jdfreder**
@minrk I think this is what you were asking for earlier..  Mostly stolen from NB.

### CODE REVIEW

File: setup.py [hunk 1]
```diff
@@ -32,6 +32,61 @@
 
 from distutils.core import setup
 
+class NPM(Command):
```
Comment (minrk) [inline setup.py]: Command hasn't been imported.

Comment (jdfreder) [inline setup.py]: Thanks, travis w/o local test ftw

**Comment by minrk** [comment]
The missing piece here is now implying css, jsdeps before sdist, develop, and build_py.

File: setup.py [hunk 2]
```diff
@@ -29,8 +29,85 @@
...
+from os.path import join as pjoin
+from distutils.core import setup, Command
+from distutils.command.build_py import build_py
+from distutils.command.sdist import sdist
+
+def css_js_prerelease(command, strict=False):
+    """decorator for building minified js/css prior to another command"""
+    class DecoratedCommand(command):
+        def run(self):
+            self.distribution.run_command('jsversion')
```
Comment (minrk) [inline setup.py]: copy/paste - jsversion isn't defined here.

File: setup.py [hunk 3]
```diff
@@ -74,8 +151,19 @@
...
+        sdist: css_js_prerelease(sdist, strict=True),
+        css: CompileCSS,
+        jsdeps: NPM
+    },
 )
 
+if 'setuptools' in sys.modules:
+    # setup.py develop should check for submodules
+    from setuptools.command.develop import develop
+    setup_args['cmdclass']['develop'] = css_js_prerelease(develop)
```
Comment (minrk) [inline setup.py]: develop should be strict, I think.

**Comment by minrk** [comment]
@jdfreder I've rebased this and added a commit in requirements.txt, and the tests are passing. Can you reset this to the state of https://github.com/minrk/ipywidgets/tree/say-no-to-gulp

?

**Comment by jdfreder** [comment]
I see you already opened a PR :)

**Comment by minrk** [comment]
I just wanted to be sure the tests were passing, since I didn't trust it on my own branch. Either way, gulp is gone, yay!

**Comment by jdfreder** [comment]
> Either way, gulp is gone, yay!

Hazzah!

**Pull request closed without merging**